contracts/xasset-orchestrator/src/orchestrator.rs

```diff
@@ -0,0 +1,165 @@
+    /// and to get the asset symbol from the contract address.
+    /// The key is the asset symbol, the value is the asset contract address.
+    assets: PersistentMap<String, Address>,
+}
+
+#[subcontract]
+pub trait IsOrchestratorTrait {
+    fn init(&mut self, xlm_sac: Address, xlm_contract: Address) -> Result<(), Error>;
+    #[allow(clippy::too_many_arguments)]
+    fn deploy_asset_contract(
+        &mut self,
+        asset_contract: Address,
+        pegged_asset: Symbol,
+        min_collat_ratio: u32,
+        name: String,
+        symbol: String,
+        decimals: u32,
+        annual_interest_rate: u32,
+    ) -> Result<Address, Error>;
+    fn get_asset_contract(&self, asset_symbol: String) -> Result<Address, Error>;
+    // Manually set an asset symbol to an existing contract address
+    fn set_asset_contract(
+        &mut self,
+        asset_symbol: String,
+        asset_contract: Address,
+    ) -> Result<(), Error>;
+    // Manually set an existing asset symbol to an existing contract address. Dangerous!
+    // This should only be used when needing to update an existing symbol's contract.
+    fn set_existing_asset_contract(
+        &mut self,
+        asset_symbol: String,
+        asset_contract: Address,
+    ) -> Result<(), Error>;
+}
+
+impl IsOrchestratorTrait for Storage {
+    fn init(&mut self, xlm_sac: Address, xlm_contract: Address) -> Result<(), Error> {
+        Contract::require_auth();
+        self.xlm_sac.set(&xlm_sac);
+        self.xlm_contract.set(&xlm_contract);
+        Ok(())
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn deploy_asset_contract(
+        &mut self,
+        asset_contract: Address,
+        pegged_asset: Symbol,
+        min_collat_ratio: u32,
+        name: String,
+        symbol: String,
+        decimals: u32,
+        annual_interest_rate: u32,
+    ) -> Result<Address, Error> {
+        let env = env();
+        Contract::require_auth();
+        // Check if the asset contract is already deployed
+        if self.assets.has(symbol.clone()) {
+            return Err(Error::AssetAlreadyDeployed);
+        }
+
+        // Upload the xasset wasm to prep for deploy
+        let wasm_hash = env.deployer().upload_contract_wasm(xasset::WASM);
+
+        // Deploy the contract, salting with the symbol
+        let deployed_contract = create_contract(env, &wasm_hash, symbol.clone());
+
+        let contract_address = deployed_contract.map_err(|_| Error::InitFailed)?;
+        // Create a client instance for this contract so we can initialize it
+        let client = xasset::Client::new(env, &contract_address);
+
+        // Initialize the contract
+        client.cdp_init(
+            &self.xlm_sac.get().unwrap(),
+            &self.xlm_contract.get().unwrap(),
+            &asset_contract,
+            &pegged_asset,
+            &min_collat_ratio,
+            &name,
+            &symbol,
+            &decimals,
+            &annual_interest_rate,
```

Comment:
This might be more readable as a struct.

Reply:
`cdp_init` is in the xasset contract, and since I'm trying not to futz with that right now, I'll leave this